### PR TITLE
Update error messages to be consistent across data sources

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/4360

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/googlesheets_client.bal [hunk 1]
```diff
@@ -240,12 +240,9 @@ public isolated client class GoogleSheetsClient {
...
         if rows.length() == 0 {
-            if key is map<anydata> {
-                return <persist:NotFoundError>error(string `Not found: ${self.generateKeyArrayString(self.keyFields, key)}`);
-            } else {
-                return <persist:NotFoundError>error(string `Not found: ${key.toString()}`);
-            }
-        } else if rows.length() > 1 {
+            return persist:getNotFoundError(self.entityName, key);
+        }
+        else if rows.length() > 1 {
```
Comment (daneshk) [inline ballerina/googlesheets_client.bal]:
```suggestion
        } else if rows.length() > 1 {
```

File: ballerina/googlesheets_client.bal [hunk 2]
```diff
@@ -259,7 +256,9 @@ public isolated client class GoogleSheetsClient {
                 } else if (key is int|string|decimal|float) {
                     values.push(key);
                 }
-            } else if !updateRecord.hasKey(entityKey) && self.keyFields.indexOf(entityKey) == () {
+            }
+
+            else if !updateRecord.hasKey(entityKey) && self.keyFields.indexOf(entityKey) == () {
```
Comment (daneshk) [inline ballerina/googlesheets_client.bal]:
```suggestion
            } else if !updateRecord.hasKey(entityKey) && self.keyFields.indexOf(entityKey) == () {
```

File: ballerina/googlesheets_client.bal [hunk 3]
```diff
@@ -268,17 +267,21 @@ public isolated client class GoogleSheetsClient {
                         return <persist:Error>error(value.message());
                     }
                     values.push(value);
-                } else {
+                }
+
+                else {
```
Comment (daneshk) [inline ballerina/googlesheets_client.bal]:
```suggestion
                } else {
```

Comment (kaneeldias) [reply ballerina/googlesheets_client.bal]: Fixed in [6af274a](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/15/commits/6af274ac3b3c8e6595db3232d142746015cd4ef0)

File: ballerina/googlesheets_client.bal (hunk 2 above)
Comment (kaneeldias) [reply ballerina/googlesheets_client.bal]: Fixed in [6af274a](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/15/commits/6af274ac3b3c8e6595db3232d142746015cd4ef0)

File: ballerina/googlesheets_client.bal (hunk 1 above)
Comment (kaneeldias) [reply ballerina/googlesheets_client.bal]: Fixed in [6af274a](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/15/commits/6af274ac3b3c8e6595db3232d142746015cd4ef0)

**Comment by kaneeldias** [comment]
Closing PR as we need to merge changes to `dev` branch first in order for the workflows to complete.

**Pull request closed without merging**